base/strings/string.jl

```diff
@@ -438,3 +449,25 @@ function repeat(s::String, r::Integer)
     end
     return out
 end
+
+"""
+    repeat(c::Char, r::Integer) -> String
+
+Repeat a character `r` times.
+
+# Examples
+```jldoctest
+julia> repeat('A', 3)
+"AAA"
+```
+"""
+function repeat(c::Char, r::Integer)
+    r < 0 && throw(ArgumentError("can't repeat a character $r times"))
```

Comment:
Put the `r < 0` check inside the `isascii` check, since `repeat(string(c), r)` also does this check; otherwise you are doing the same check twice for non-ascii chars.